docs/RFCS: redact statement diagnostics bundles

michae2 opened the pull request:
**docs/RFCS: redact statement diagnostics bundles**

Epic: CRDB-19756

Release note: None

---

**sql: add REDACT flag to EXPLAIN ANALYZE (DEBUG)**

And also use it to hide constants in the statements in statement.txt
and trace.txt.

Release note: None

Epic: CRDB-19756

michae2 committed:
sql: add REDACT flag to EXPLAIN ANALYZE (DEBUG)

And also use it to hide constants in the statements in statement.txt
and trace.txt.

Release note: None

Epic: CRDB-19756

michae2 committed:
docs/RFCS: redact statement diagnostics bundles

Epic: CRDB-19756

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/92306)
<!-- Reviewable:end -->

michae2 committed:
sql: redact trace.txt when using EXPLAIN ANALYZE (DEBUG, REDACT)

(Ideally we would be setting spans to redactable rather than the entire
tracer.)

Release note: None

Epic: CRDB-19756

michae2 committed:
hide histograms

michae2 committed:
better trace redaction
